File: searches/single_element_in_a_sorted_array_binary_search.py
class Solution:
    def singleNonDuplicate(self, nums: List[int]) -> int:
        left = 0
        right = len(nums) - 1

        while left < right:
            mid = (left + right) // 2
            
            if mid % 2 == 1:  # if mid is odd
                mid -= 1
            
            if nums[mid] == nums[mid + 1]:
                left = mid + 2
            else:
                right = mid
        
        return nums[left]

File: searches/single_element_in_a_sorted_array_binary_search.rb
def single_non_duplicate(nums)
    left  = 0
    right = len(nums) - 1

    while left < right
        mid = (left + right) // 2
        if mid % 2 == 1:
                mid -= 1

        if nums[mid] == nums[mid + 1]
            left = mid + 2
        else
            right = mid - 2
        end
    end

    nums[left]
end